pkg/trtl/honu.go

```diff
@@ -111,8 +112,74 @@ func (h *HonuService) Iter(ctx context.Context, in *pb.IterRequest) (out *pb.Ite
 	return nil, status.Error(codes.Unimplemented, "not implemented")
 }
 
-func (h *HonuService) Batch(stream pb.Trtl_BatchServer) (err error) {
-	return status.Error(codes.Unimplemented, "not implemented")
+// Batch is a client-side streaming request to issue multiple commands, usually Put and Delete.
+func (h *HonuService) Batch(stream pb.Trtl_BatchServer) error {
+	out := &pb.BatchReply{}
+	for {
+		// Read the next request from the stream.
+		in, err := stream.Recv()
+		if err == io.EOF {
+			return stream.SendAndClose(out)
+		}
+		if err != nil {
+			return status.Error(codes.Internal, err.Error())
+		}
```

Comment:
These are the loop terminal conditions, and it appears that "successful" streams require the client to close the stream and send an `io.EOF` error. This pattern could work for detecting when the batch is complete, but I'm not sure how to implement it on the client side. Have you tested this functionality with a local server?

Reply:
We do have a story for adding Batch and Cursor tests so I didn't want to overlap the scope of this too much, but I do think it's worth it to add a simple test to this PR to figure out what the client side would look like.

Reply:
Excellent, thank you; and thank you for finding the `CloseAndRecv` function, I had no idea about that but it makes a lot of sense in client-side and server-side streaming!